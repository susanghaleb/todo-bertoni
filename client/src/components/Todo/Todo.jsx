import React, { memo, Fragment } from "react";
import PropTypes from "prop-types";

import { makeStyles } from "@material-ui/core/styles";
import ListItem from "@material-ui/core/ListItem";
import Divider from "@material-ui/core/Divider";
import ListItemText from "@material-ui/core/ListItemText";
import Checkbox from "@material-ui/core/Checkbox";
import Button from "@material-ui/core/Button";
import DeleteIcon from "@material-ui/icons/Delete";

const Todo = ({ task, onDelete, onCheck }) => {
  const useStyles = makeStyles(theme => ({
    button: {
      margin: theme.spacing(1)
    },
    rightIcon: {
      marginLeft: theme.spacing(1)
    },
    textDecoration: {
      textDecoration: "line-through"
    }
  }));
  const classes = useStyles();

  const { label, done } = task;
  const handleDelete = () => onDelete(task);
  const handleChange = () => onCheck({ ...task, done: !task.done });

  return (
    <Fragment>
      <ListItem alignItems="center">
        <Checkbox checked={done} onChange={handleChange} color="primary" />
        <ListItemText
          primary={label}
          className={done ? classes.textDecoration : ""}
        />
        <Button
          classkey="delete"
          variant="contained"
          color="secondary"
          className={classes.button}
          onClick={handleDelete}
        >
          Delete
          <DeleteIcon className={classes.rightIcon} />
        </Button>
      </ListItem>
      <Divider />
    </Fragment>
  );
};

Todo.propTypes = {
  task: PropTypes.shape({
    label: PropTypes.string,
    done: PropTypes.bool
  }).isRequired,
  onDelete: PropTypes.func.isRequired,
  onCheck: PropTypes.func.isRequired
};

export default memo(Todo);
