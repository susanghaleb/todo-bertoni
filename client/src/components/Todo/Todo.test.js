import React from "react";
import renderer from "react-test-renderer";
import Todo from "./Todo";

it("renders correctly", () => {
  const taskMock = {
    label: "ONE",
    id: 87679,
    done: false
  };

  const mockFunction = jest.fn();
  const tree = renderer
    .create(
      <Todo task={taskMock} onCheck={mockFunction} onDelete={mockFunction} />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
