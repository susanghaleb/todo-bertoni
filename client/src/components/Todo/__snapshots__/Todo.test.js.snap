// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <li
    className="MuiListItem-root MuiListItem-gutters"
    disabled={false}
  >
    <span
      aria-disabled={false}
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary PrivateSwitchBase-root-21 MuiCheckbox-root MuiCheckbox-colorPrimary"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={null}
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
        </svg>
        <input
          checked={false}
          className="PrivateSwitchBase-input-24"
          data-indeterminate={false}
          onChange={[Function]}
          type="checkbox"
        />
      </span>
    </span>
    <div
      className="MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary"
      >
        ONE
      </span>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root makeStyles-button-1 MuiButton-containedSecondary MuiButton-contained"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Delete
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root makeStyles-rightIcon-2"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
          />
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
        </svg>
      </span>
    </button>
  </li>,
  <hr
    className="MuiDivider-root"
  />,
]
`;
